### ISSUE TITLE
Could not create user: unknown_error

### ISSUE THREAD

**Issue by BlackSmith** [keycloak/keycloak#21149]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

account/ui

### Describe the bug

 I can not create local user, when the LDAP Federation servers are enabled.
When I try to create new local user account, I get error message “Could not create user: unknown_error”.

I use 5 LDAP Federation servers, everything (except this) works well. The users can log in, I can search them, …
However, when I try to create a local user, it fails. Workaround is disabling all federation servers, create user and enabling all servers.

I have tested this on versions 21.1 and 20.0.5. The last time, when I successfully create local user was November 2022 (I guess it was version 19.x).

System error log:
```
2023-06-22 10:59:38,172 INFO  [org.keycloak.storage.ldap.LDAPIdentityStoreRegistry] (executor-thread-172) Creating new LDAP Store for the LDAP storage provider: 'cz.xxx.eu', LDAP Configuration: {serverPrincipal=HTTP/keycloak.cz.xxx.eu@CZ.XXX.EU], fullSyncPeriod=[604800], pagination=[true], startTls=[false], sersDn=[DC=cz,DC=xxx,DC=eu], connectionPooling=[true], cachePolicy=[DEFAULT], useKerberosForPasswordAuthentication=[true], importEnabled=[true], enabled=[true], bindDn=[cn=ldap binder,ou=ldapbinder,ou=user,ou=mis,dc=cz,dc=xxx,dc=eu], usernameLDAPAttribute=[sAMAccountName], changedSyncPeriod=[86400], lastSync=[1687392697], vendor=[ad], uuidLDAPAttribute=[objectGUID], connectionUrl=[ldap://cz.xxx.eu:389], allowKerberosAuthentication=[false], syncRegistrations=[false], authType=[simple], debug=[true], searchScope=[2], useTruststoreSpi=[ldapsOnly], keyTab=[/etc/krb5.keytab], usePasswordModifyExtendedOp=[false], trustEmail=[true], priority=[0], kerberosRealm=[CZ.XXX.EU], userObjectClasses=[person, organizationalPerson, user], rdnLDAPAttribute=[cn], editMode=[UNSYNCED], readTimeout=[30000], validatePasswordPolicy=[false], batchSizeForSync=[1000]}, binaryAttributes: []


2023-06-22 10:59:42,175  ERROR [org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager] (executor-thread-172) Could not query server using DN [DC=cz,DC=xxx,DC=eu] and filter [(&(sAMAccountName=test@test.cz)(objectclass=person)(objectclass=organizationalPerson)(objectclass=user))]: javax.naming.InterruptedNamingException: Interrupted during LDAP operation

	at java.naming/com.sun.jndi.ldap.Connection.readReply(Connection.java:446)
	at java.naming/com.sun.jndi.ldap.LdapClient.ldapBind(LdapClient.java:365)
	at java.naming/com.sun.jndi.ldap.LdapClient.authenticate(LdapClient.java:214)
	at java.naming/com.sun.jndi.ldap.LdapCtx.connect(LdapCtx.java:2896)
	at java.naming/com.sun.jndi.ldap.LdapCtx.<init>(LdapCtx.java:348)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getLdapCtxFromUrl(LdapCtxFactory.java:229)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getUsingURL(LdapCtxFactory.java:189)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getUsingURLs(LdapCtxFactory.java:247)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getLdapCtxInstance(LdapCtxFactory.java:154)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getInitialContext(LdapCtxFactory.java:84)
	at java.naming/javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:732)
	at java.naming/javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:305)
	at java.naming/javax.naming.InitialContext.init(InitialContext.java:236)
	at java.naming/javax.naming.ldap.InitialLdapContext.<init>(InitialLdapContext.java:154)
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPContextManager.createLdapContext(LDAPContextManager.java:80)
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPContextManager.getLdapContext(LDAPContextManager.java:101)
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.execute(LDAPOperationManager.java:710)
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.execute(LDAPOperationManager.java:705)
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.search(LDAPOperationManager.java:253)
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPIdentityStore.fetchQueryResults(LDAPIdentityStore.java:279)
	at org.keycloak.storage.ldap.idm.query.internal.LDAPQuery.getResultList(LDAPQuery.java:172)
	at org.keycloak.storage.ldap.idm.query.internal.LDAPQuery.getFirstResult(LDAPQuery.java:183)
	at org.keycloak.storage.ldap.LDAPStorageProvider.loadLDAPUserByUsername(LDAPStorageProvider.java:837)
	at org.keycloak.storage.ldap.LDAPStorageProvider.getUserByUsername(LDAPStorageProvider.java:533)
	at org.keycloak.storage.UserStorageManager.lambda$getUserByUsername$16(UserStorageManager.java:334)
	at org.keycloak.utils.ServicesUtils.lambda$timeBoundOne$1(ServicesUtils.java:84)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.SortedOps$RefSortingSink.end(SortedOps.java:400)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:528)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
	at org.keycloak.storage.UserStorageManager.getUserByUsername(UserStorageManager.java:334)
	at org.keycloak.models.cache.infinispan.UserCacheSession.getUserByUsername(UserCacheSession.java:269)
	at org.keycloak.services.resources.admin.UsersResource.createUser(UsersResource.java:133)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:192)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:152)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:183)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:152)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:183)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:141)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:32)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:151)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:82)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:42)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:173)
	at io.vertx.ext.web.impl.RoutingContextWrapper.next(RoutingContextWrapper.java:200)
	at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:84)
	at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:71)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:173)
	at io.vertx.ext.web.impl.RoutingContextWrapper.next(RoutingContextWrapper.java:200)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$6.handle(VertxHttpRecorder.java:430)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$6.handle(VertxHttpRecorder.java:408)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:173)
	at io.vertx.ext.web.impl.RoutingContextWrapper.next(RoutingContextWrapper.java:200)
	at org.keycloak.quarkus.runtime.integration.web.QuarkusRequestFilter.lambda$createBlockingHandler$0(QuarkusRequestFilter.java:82)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
2023-05-10 08:10:28,794 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-65) Uncaught server error: org.keycloak.models.ModelException: LDAP Query failed
	at org.keycloak.storage.ldap.idm.query.internal.LDAPQuery.getResultList(LDAPQuery.java:176)
	at org.keycloak.storage.ldap.idm.query.internal.LDAPQuery.getFirstResult(LDAPQuery.java:183)
	at org.keycloak.storage.ldap.LDAPStorageProvider.loadLDAPUserByUsername(LDAPStorageProvider.java:837)
	at org.keycloak.storage.ldap.LDAPStorageProvider.getUserByUsername(LDAPStorageProvider.java:533)
	at org.keycloak.storage.UserStorageManager.lambda$getUserByUsername$16(UserStorageManager.java:334)
	at org.keycloak.utils.ServicesUtils.lambda$timeBoundOne$1(ServicesUtils.java:84)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.SortedOps$RefSortingSink.end(SortedOps.java:400)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:528)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
	at org.keycloak.storage.UserStorageManager.getUserByUsername(UserStorageManager.java:334)
	at org.keycloak.models.cache.infinispan.UserCacheSession.getUserByUsername(UserCacheSession.java:269)
	at org.keycloak.services.resources.admin.UsersResource.createUser(UsersResource.java:133)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:192)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:152)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:183)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:152)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:183)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:141)
	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:32)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:151)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:82)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:42)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:173)
	at io.vertx.ext.web.impl.RoutingContextWrapper.next(RoutingContextWrapper.java:200)
	at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:84)
	at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:71)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:173)
	at io.vertx.ext.web.impl.RoutingContextWrapper.next(RoutingContextWrapper.java:200)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$6.handle(VertxHttpRecorder.java:430)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$6.handle(VertxHttpRecorder.java:408)
	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:173)
	at io.vertx.ext.web.impl.RoutingContextWrapper.next(RoutingContextWrapper.java:200)
	at org.keycloak.quarkus.runtime.integration.web.QuarkusRequestFilter.lambda$createBlockingHandler$0(QuarkusRequestFilter.java:82)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.keycloak.models.ModelException: Querying of LDAP failed org.keycloak.storage.ldap.idm.query.internal.LDAPQuery@81dc261
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPIdentityStore.fetchQueryResults(LDAPIdentityStore.java:289)
	at org.keycloak.storage.ldap.idm.query.internal.LDAPQuery.getResultList(LDAPQuery.java:172)
	... 70 more
Caused by: javax.naming.InterruptedNamingException: Interrupted during LDAP operation
	at java.naming/com.sun.jndi.ldap.Connection.readReply(Connection.java:446)
	at java.naming/com.sun.jndi.ldap.LdapClient.ldapBind(LdapClient.java:365)
	at java.naming/com.sun.jndi.ldap.LdapClient.authenticate(LdapClient.java:214)
	at java.naming/com.sun.jndi.ldap.LdapCtx.connect(LdapCtx.java:2896)
	at java.naming/com.sun.jndi.ldap.LdapCtx.<init>(LdapCtx.java:348)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getLdapCtxFromUrl(LdapCtxFactory.java:229)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getUsingURL(LdapCtxFactory.java:189)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getUsingURLs(LdapCtxFactory.java:247)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getLdapCtxInstance(LdapCtxFactory.java:154)
	at java.naming/com.sun.jndi.ldap.LdapCtxFactory.getInitialContext(LdapCtxFactory.java:84)
	at java.naming/javax.naming.spi.NamingManager.getInitialContext(NamingManager.java:732)
	at java.naming/javax.naming.InitialContext.getDefaultInitCtx(InitialContext.java:305)
	at java.naming/javax.naming.InitialContext.init(InitialContext.java:236)
	at java.naming/javax.naming.ldap.InitialLdapContext.<init>(InitialLdapContext.java:154)
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPContextManager.createLdapContext(LDAPContextManager.java:80)
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPContextManager.getLdapContext(LDAPContextManager.java:101)
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.execute(LDAPOperationManager.java:710)
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.execute(LDAPOperationManager.java:705)
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPOperationManager.search(LDAPOperationManager.java:253)
	at org.keycloak.storage.ldap.idm.store.ldap.LDAPIdentityStore.fetchQueryResults(LDAPIdentityStore.java:279)
	... 71 more
```
![image](https://github.com/keycloak/keycloak/assets/192467/91fe78a5-8a0b-40c1-ae76-62db29ad9980)

### Version

21.1.1

### Expected behavior

No error and successful create local user or if possible, better error message.

### Actual behavior

error message saying nothing 

### How to Reproduce?

Enable user federation and try to create local keycloak user.

**Comment by ssilvert** [comment]
@BlackSmith I changed this to `area/admin/ui`.  That's what you meant, right?

Give than that error happens on the server side, it's probably not a UI issue at all.  But one thing we can do is to install the old admin console and see for sure.  Since this was working on an old version, perhaps the old admin console will tell us if it's a bug in the new one.

Please install the old admin console from here and let us know how it goes:
https://github.com/ssilvert/old-keycloak-admin-console

**Comment by BlackSmith** [comment]
@ssilvert  I'm not sure that I did it correctly.
I have unziped  [old-admin-console.zip](https://github.com/ssilvert/old-keycloak-admin-console/blob/main/old-admin-console.zip) into `/opt/keycloak/themes` and restarted keycloak.
In my REALM > realm settings > themes, I have set `Admin UI theme` to `keycloak`. 
But I can not see any design changes and creating of new user still does not work. If I choose any template from dropdown, the design  does not change.
I guess, I do something wrong. :(

**Comment by cbioley** [comment]
@BlackSmith Are you running your Keycloak instance in Kubernetes by any chance? 😬

**Comment by BlackSmith** [comment]
@cbioley No, by docker-compose.

**Comment by cbioley** [comment]
@BlackSmith 

Ok. 

We have a running version 15-ish in production for years and everything is working fine. We tried to install newer versions (Keycloak 19, 20 and 21) this week and we kept on having the same exact error you are experiencing (Interrupted during LDAP operation) even with read operations.

Everything is working fine with Docker.

Sorry to have stepped in your issue then 🙈

**Comment by BlackSmith** [comment]
@cbioley 
I'm not sure that this is important, but  I have two containers/instances of the Keycloak with the same db backend. (One for intranet and second for internet). It can be relation to k8s configuration.

I have created this configuration a few years ago and everything worked well. Except this now.

**Comment by ssilvert** [comment]
> @ssilvert I'm not sure that I did it correctly. I have unziped [old-admin-console.zip](https://github.com/ssilvert/old-keycloak-admin-console/blob/main/old-admin-console.zip) into `/opt/keycloak/themes` and restarted keycloak. In my REALM > realm settings > themes, I have set `Admin UI theme` to `keycloak`. But I can not see any design changes and creating of new user still does not work. If I choose any template from dropdown, the design does not change. I guess, I do something wrong. :(

@BlackSmith Though they look a little similar, you should definitely be able to tell the difference between old and new admin console.  For one thing, the lefthand menu navigation is different.

Maybe you set the theme to `keycloak` on the wrong realm?

**Comment by ssilvert** [comment]
Given that we are seeing a server-side error, it's unlikely that this is a UI bug.  Switching to the core team.

**Comment by douglaspalmer** [comment]
@BlackSmith Does increasing the LDAP read timeout help with this issue? From the stack trace it looks like you're getting a timeout.

**Comment by pedroigor** [comment]
@BlackSmith Does it work if you have a single LDAP provider enabled at runtime?

**Comment by BlackSmith** [comment]
@pedroigor No. It does not.  I got this error always on the first LDAP server in the priority order. I can create the new user only when I disable all LDAP provides.

Maybe can help you my docker-compose.yaml file
```yaml
version: '3'

services:
  postgres:
    image: postgres:10-bullseye
    labels:
      - "app=keycloak"
    volumes:
      - ./data/pg:/var/lib/postgresql/data
    environment:
      POSTGRES_DB: keycloak
      POSTGRES_USER: keycloak
      POSTGRES_PASSWORD: xxx
    logging:
      driver: 'journald'
      options:
        labels: 'app'

  keycloak_public:
    image: quay.io/keycloak/keycloak:latest
    container_name: keycloak_public
    labels:
      - "app=keycloak"
    ports:
      - "127.0.0.1:18001:8080"
    environment:
      TZ: Europe/Prague
      KEYCLOAK_ADMIN: admin
      KEYCLOAK_ADMIN_PASSWORD: xxx
      KC_DB: postgres
      KC_DB_DATABASE: keycloak
      KC_DB_URL_HOST: postgres
      KC_DB_USERNAME: keycloak
      KC_DB_SCHEMA: public
      KC_DB_PASSWORD: xxx
      KC_HOSTNAME: keycloak.public.company.cz
      KC_HTTP_ENABLED: "true"
      KC_METRICS_ENABLED: "true"
      KC_PROXY: edge
      JAVA_OPTS_APPEND: -Dquarkus.transaction-manager.default-transaction-timeout=3600000
    command: start --auto-build --http-relative-path /auth --log-level="INFO,org.keycloak.storage.ldap:DEBUG"
    depends_on:
      - postgres
    restart: always
    logging:
      driver: 'journald'
      options:
        labels: 'app'

  keycloak_private:
    image: quay.io/keycloak/keycloak:latest
    container_name: keycloak_private
    labels:
      - "app=keycloak"
    ports:
      - "127.0.0.1:18000:8080"
    environment:
      TZ: Europe/Prague
      KEYCLOAK_ADMIN: admin
      KEYCLOAK_ADMIN_PASSWORD: xxx
      KC_DB: postgres
      KC_DB_DATABASE: keycloak
      KC_DB_URL_HOST: postgres
      KC_DB_USERNAME: keycloak
      KC_DB_SCHEMA: public
      KC_DB_PASSWORD: xxx
      KC_HOSTNAME: keycloak.private.company.eu
      KC_HTTP_ENABLED: "true"
      KC_METRICS_ENABLED: "true"
      KC_PROXY: edge
      JAVA_OPTS_APPEND: -Dquarkus.transaction-manager.default-transaction-timeout=3600000
#    volumes:
#      - ./themes:/opt/keycloak/themes
    command: start --auto-build --http-relative-path /auth --log-level="INFO,org.keycloak.storage.ldap:DEBUG"
    depends_on:
      - postgres
    restart: always
    logging:
      driver: 'journald'
      options:
        labels: 'app'
```

I have tested it in a different company, I have there the simplest configuration with the same version (one keycloak instance + one LDAP server) and the problem does not raise up.

**Comment by pedroigor** [comment]
The fact it works in a different environment indicates something is wrong with yours?

I just run your container specification and configured a single LDAP provider (ApacheDS). I was able to create users and authenticate.

One thing I noted in your container specification is the usage of `auto-build`. You don't need to set it anymore because the server defaults to "auto-build" by default. See https://www.keycloak.org/server/configuration#_starting_keycloak and https://www.keycloak.org/server/configuration#_optimize_the_keycloak_startup. Other than that, LGTM.

There are two things I can think of:

* DNS. Perhaps you can try using the LDAP server IP address rather than its domain name to see how it goes
* Amount of data in LDAP directory. As you are facing this in a production-grade LDAP directory, it might happen that when querying the directory Keycloak is taking too much time to get a response back. However, you already tried setting a time out so I would ask you to try connecting to a fresh and small LDAP directory structure to see how it goes.

Other than that, I'm clueless ...

**Comment by BlackSmith** [comment]
@pedroigor  Thank you. 

I have taken a few days for  testing and this is my results:
- I swapped the LDAP domain names to IP, and it helped. The huge ldap servers have got many mirrors and unfortunately about 30% of them are offline. When the keycloak chooses one of them and this server is offline, the keycloak does not try any others.  

- I have changed read & connection time out limit. At the beginning, I didn't realize, when the timeout works for synchronization process it does not mean it works for search. The synchronization process gets data from servers continuously from the beginning, however the search process has to wait for results (the timeout has to be much bigger).

I think that some better logic of choosing IP from DNS records can help many users.  For example, it could use the last functional server (IP) or try to ping them. But I know, it should be placed in a separate ticket. :)

For now, I take this ticket as solved. 
Thank you all for help.
